### ISSUE TITLE
slow scrolling jumps backward

### ISSUE THREAD

**Issue by eritain** [GoogleChromeLabs/text-app#303]
Quick scrolling is able to move through documents -- ballistic scrolling, where the document continues moving after my fingers have left the touchpad. But below a certain speed, scrolling is just spinning my wheels: after scrolling through about a screenful of content, the screen jumps back to approximately the same line it started on. Sometimes further than where it started.

The jump is reflected across the UI: the box in the scroll bar pops back to its original position as though on a spring, and the line numbers and line content jump back too. 

The jumping-back phenomenon is also visible in the scroll box during the later portion of long ballistic scrolls.

**Comment by ste4es** [comment]
Same issue.  Seems to have coincided with a recent ChromeOS update.
Linux localhost 3.10.18 #1 SMP Wed Feb 22 23:32:23 PST 2017 x86_64 Intel(R) Celeron(R) CPU N2840 @ 2.16GHz GenuineIntel GNU/Linux

**Comment by danhab99** [comment]
This is beyond reason, I have to scroll slowly with the arrow keys. I have no idea how this extension even works to fix it myself.

**Comment by murthag11** [comment]
Same here. I'd love this to get fixed, it's the nicest Editor on ChromeOS imo.
Luckily, i can scroll with the Touchscreen (Pixel 2025), but sometimes when I click in a line to write there, it also jumps somewhere and only refocuses when I start typing.

**Comment by Sputternik** [comment]
I'd appreciate a fix for this too. I agree with murthag11, this IS the nicest, and cleanest editor on ChromeOS. I am using Carat for the time being, but the feature implementation on Editor is better/cleaner, and Caret can't print without line numbers (at least, I have not had luck determining js code to suppress them on the printout). So I much prefer Editor. A big Thank You to the developer!

**Comment by synthstow** [comment]
I have this too, extremely irritating

**Comment by josephrocca** [comment]
This editor is basically unusable with this bug. Any hope of this being fixed any time soon @eterevsky? Is it being maintained? Such a nice little app otherwise - would love to keep it as my default editor.

**Comment by danhab99** [comment]
@josephrocca I've stopped completely. I new use Caret!

**Comment by bawden** [comment]
The most recent ChromOS upgrade installed this editor as if Google thinks it actually works, but the scrolling problem make it almost worthless.  What were they thinking!?

**Comment by SiebertBrandon** [comment]
Confirmed this is occurring for me as well. The view bounces back when scrolling with the track pad. It seems to be independent of what line currently has the cursor.

Chrome OS Version 64.0.3261.0 (Official Build) dev (64-bit)
Acer Chromebook 14
Text App version 0.5.68

**Comment by DefecateRainbows** [comment]
Happens here too.

**Comment by ammaarpatel99** [comment]
This is a real problem and happens unless scrolling really fast, which is pretty useless as scrolling that fast only allows you to go from the top of a website to the bottom, and nowhere in between.

**Comment by JerryWiltz** [comment]
Happens here too. If I scroll fast it, gets down to the bottom. But if I scroll slow, it bounces back up. Very frustrating.

**Comment by tomohaha3** [comment]
bump

**Comment by ButteryRolls** [comment]
bump

**Comment by SHADOWELITE7** [comment]
you know they aren't going to update the text editor

**Comment by DefecateRainbows** [comment]
Well this is GitHub, for god's sake.
I'll make a fork of the code, when I hate time.
There might already be one.

**Comment by SHADOWELITE7** [comment]
@DefecateRainbows what 3d modeller do you use? it looks nice tbh

**Comment by DefecateRainbows** [comment]
Feel free to start your own in the meantime, it'll be some time before my fork is finished.
Just Blender!
... did you google my profile picture to see if it was original lol?
I actually put it on opengameart
https://opengameart.org/content/low-poly-knight

**Comment by SHADOWELITE7** [comment]
thx

**Comment by tac11tac** [comment]
I am having this issue, also.  It is particularly obnoxious when using a scroll mouse.  I've not seen it with my touchpad or touchscreen.

**Comment by Mythius** [comment]
This happens for me as well.
Only thing I can do is use right hand side scroll bar.

**Comment by KidneyPunch** [comment]
This is super frustrating to have jump all over the place, then try to find where your cursor is.  I wasn't a huge fan of Carat, but if someone has another editor they recommend I'd love to try it out.

**Comment by GitBoudewijn** [comment]
This issue makes this app almost unusable with large files.

**Comment by oliversiegel** [comment]
Me too

**Comment by BugsNash** [comment]
Can reliably repro with both mouse and touchpad on chromeos. When scrolling slowly from the top at around line 38 it jumps back to around line 10.

**Comment by BugsNash** [comment]
This seems to be caused by the CodeMirror library we're using. It seems the latest version of CodeMirror (5.39.0) doesn't have this issue, so updateing the CodeMirror library should fix the problem.

**Comment by GitBoudewijn** [comment]
Any idea when this will be updated? It would be nice to be able to use this app again after more than a year.

**Comment by BugsNash** [comment]
Hopefully on the order of weeks? Working on it now, there are a few issues with hard coded links to stuff that no longer exists and changes we've made to the version of the library that we use.

This will also require some QA testing (which I will send to another team) before uploading the new version of CodeMirror to make sure I didn't miss anything that is broken by the update (I've found 1 thing that breaks with the new version so far), so it depends how long that will take.
